src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,34 @@
+import {
+    Button,
+    HStack,
+    Input,
+    useNumberInput
+  } from '@chakra-ui/react'
+
+type CounterProps = {
+    type: string;
+};
+
+function Counter({ type }: CounterProps) {
+    const { getInputProps, getIncrementButtonProps, getDecrementButtonProps } =
+        useNumberInput({
+            step: 1,
+            defaultValue: 0,
+            min: 0,
+            max: 10
+        })
+  
+    const inc = getIncrementButtonProps()
+    const dec = getDecrementButtonProps()
+    const input = getInputProps()
+  
+    return (
+      <HStack maxW='100px'>
+        <Button {...inc}>+</Button>
+        <Input {...input} />
+        <Button {...dec}>-</Button>
+      </HStack>
```

Comment:
Looks good, but lets utilize InputGroup that is available to use to group elements within the Input filed.


Example:
```jsx
    <InputGroup>
      <InputLeftElement>
        <Button {...inc}>+</Button>
      </InputLeftElement>
      <Input {...input} />
      <InputRightElement>
        <Button {...dec}>-</Button>
      </InputRightElement>
    </InputGroup>
```

![Screenshot 2023-08-20 at 4 13 34 PM](https://github.com/nitishkaf/featherswatch/assets/47726833/49c7e7fc-5999-4e04-9876-eddc8171e601)

Reply:
I will push this change :)